File: dataset/human/java/file_8977.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        HashSet<Integer> holeList  = new HashSet<>();

        for (int i = 0; i < m; i++){
            holeList.add(sc.nextInt());
        }

        long count = 0;
        long countOneBefore = 1;
        long countTwoBefore = 0;

        for (int i = 0 ; i < n; i++){
            if (holeList.contains(i + 1)){
                // 穴が開いている場合
                count = 0;
                countTwoBefore = countOneBefore;
                countOneBefore = count;
            }else {
                count = countOneBefore + countTwoBefore;
                if (count >= 1000000007L){
                    count = count % 1000000007L;
                }
                countTwoBefore = countOneBefore;
                countOneBefore = count;
            }
        }

        System.out.println(count);
    }

}

File: dataset/human/c/file_8977.c
#include <stdio.h>
int main(void)
{
  int n;
  int count=0;

  while(1)
    {
      scanf("%d",&n);
      if(n == 0) break;
      while(n != 1)
	{
	  if(n % 2 == 0)
	    {
	      n /= 2;
	      count++;
	    }
	  else if(n % 2 == 1)
	    {
	      n = n * 3 + 1;
	      count++;
	    }
	}
      printf("%d\n",count);
      count = 0;
    }
  return 0;
}
